app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/K9Receiver.java

```diff
@@ -33,6 +34,17 @@ public void onReceive(Context context, Intent intent) {
                 return;
             }
         }
+        if (prefs.getBoolean("notification_filter", false) && GBApplication.isRunningMarshmallowOrLater()) {
```

Comment:
The following code looks partly duplicated among the different receivers. Can you centralize this somewhere? Either in a base class or a helper method?

For example, you could add a method getGrantedInterruptionFilter() that would return the actual filter if all conditions are met, and INTERRUPTION_FILTER_ALL for any failed condition.